### DISCUSSION TITLE
AIS-catcher integration with Signal K, opencpn, Open Chart Plotter

### DISCUSSION THREAD

**Discussion by tkreyche** [jvde-github/AIS-catcher#175] [General]
Does anyone have experience integrating AIS-catcher with programs designed for seaborn operation? It should work fine but I need to sort out the details. Their documentation is a little cryptic.

Previously I've only considered using AIS-catcher as a base station. Now I'm thinking about getting a small sailboat so I can have a waterborne platform for electronic experiments (and maybe do some sailing too). 

thanks, Tom Kreyche
Seattle Washington USA

**Comment by Taxom** [comment]
I use AIS-catcher for OpenCPN on phone and PC. 
Just add <-S PORT > in config AIS-catcher and add connections for this PORT in OpenCPN.

**Comment by tkreyche** [comment]
Thanks Taxom - I find it a little confusing since there are plug-ins for Signal K that are similar to features in OpenCPN. So it's unclear to me when it's possible to just use Signal K signal integration and its UI, and opposed to adding OpenCPN or another more powerful client. It looks like the Signal K UI is web based, while OpenCPN uses an OS based UI. 

I want to have a client-server setup, so the server can host all the connected instruments, and then I can use a phone or PC based client, with only connections over wifi (or cell).

**Comment by Taxom** [comment]
I do not really understand which application you want to see ships in. I gave an example just the use of OpenCPN as a client to obtain coordinates of ships from AisCatcher. In principle, any application that is able to receive NMEA by TCP or UDP can do this.

Here's what AisCathcer sends to the port:

>!AIVDM,3,3,7,A,e2B94oCPH54M`1SwpGwwjE69@P0,2*31
!AIVDM,1,1,,B,24eG?r0u@0JENdRHc?>@Wmu`06R0,0*2F
!AIVDM,3,1,8,A,8h30ot1?0@6Ml8D9e2B94oCPH54M`0kwprwwin6AJP2Ml:PPP121IoCo,0*7D
!AIVDM,3,2,8,A,l54cd2kwuGwwjp3aTP2Ml;D9e2B94oCPH54M`0Swpvwwin6AJP2Ml<D9,0*5A
!AIVDM,3,3,8,A,e2B94oCPH54M`0owqJwwin6AJP0,2*55
!AIVDM,1,1,,A,14eGB>@000rBhe6HhNp000Ub0<0B,0*61
!AIVDM,1,1,,A,ENkaSQLHh@@@@@@@@@@@@@@@@@@=;I<E<@lMh00003vP000,2*6D

**Comment by Taxom** [comment]
Maybe it will help with signalk server:
https://openplotter.readthedocs.io/en/latest/signalk/multiplexing.html

**Comment by tkreyche** [comment]
I don't have a preferred client application yet, just trying to figure out what the options are. Thanks for the tips.

**Comment by tkreyche** [comment]
The installation using OpenPlotter works well because it include SignalK, OpenCPN and other components. It was pretty easy to get a generic USB gps working, and AIS-catcher using TCP from a different Pi. I expect installing AIS-catcher and all the OpenPlotter components on the same Pi would work, although the map in OpenCPN is not too responsive and may slow things a bit much much. The map in AIS-catcher works better but OpenCPN has the option to download raster and vector marine charts and has many other features. There is a fair amount of discussion in OpenChart docs on how to integrate AIS and AIS hardware, in particular dAISy Hat. Jasper if you read this, you may want to send the OpenPlotter people a note on how to integrate AIS-catcher using an inexpensive dongle.

**Reply by jvde-github** [reply]
Thanks. Will have a look at some of the discussions. [This](https://www.youtube.com/watch?v=SJ5eHXnsQ4g) is a pretty good trick but I believe that plugin stopped working with the latest OpenCPN version.

If your device is USB connected probably can read from it using `-e` and  multiplex it to other software via `-P/S/u`. Or indeed, you get the NMEA into AIS-catcher via a TCP client (-t) or a UDP server  (-x) and you can run multiple of these. 

If there is any GPS NMEA flowing through Catcher, it will be used to set the location of the station, so that would be useful. You can also use `OWN_MMSI` to set the location of the station from AIS data it receives for the specific MMSI.

On a ship without a SDR dongle I see limited use of AIS-catcher as the webclient is  limited and not intended for navigation. I don't expect much more functionality in the future for the client. I am looking into more offline maps though as there are free nautical maps for my area.
